openprescribing/frontend/tests/commands/test_import_measures_new.py

```diff
@@ -190,15 +215,246 @@ def test_import_measures(self):
 
-        ccgs = self.calculate_measure(
+        regtms = self.calculate_cost_based_measure(
             numerators,
             denominators,
-            'ccg',
-            PCT.objects.values_list('code', flat=True)
+            'regional_team',
+            RegionalTeam.objects.values_list('code', flat=True),
         )
-        self.validate_measure_global(mg, ccgs, 'ccg')
+        self.validate_cost_based_measure_global(mg, regtms, 'regional_team')
+        mvs = MeasureValue.objects.filter(
+            month=month,
+            practice_id__isnull=True,
+            pct_id__isnull=True,
+            stp_id__isnull=True,
+            regional_team_id__isnull=False,
+        )
+        self.assertEqual(mvs.count(), 5)
+        for mv in mvs:
+            self.validate_cost_based_measure_value(mv, regtms.loc[mv.regional_team_id])
+
+    def test_import_measures_practice_statistics(self):
+        # This test verifies the behaviour of import_measures for measures
+        # whose denominator involves practice statistics, which may be null for
+        # a given practice in a given month.  It uses the coproxamol measure.
+        # See #1520 for background.
+
+        # Generate random prescribing data.  This data is never saved to the
+        # database.
+        presentations = [
+            ('0407010Q0AAAAAA', 'Co-Proxamol_Tab 32.5mg/325mg'),  # relevant
+            ('0407010AAAAAAAA', 'Aspirin/Caffeine_Tab 500mg/32mg'),  # irrelevant
+        ]
+
+        prescribing_rows = []
+
+        for practice in Practice.objects.all():
+            for month in [7, 8]:
+                timestamp = '2018-0{}-01 00:00:00 UTC'.format(month)
+
+                for bnf_code, bnf_name in presentations:
+                    items = randint(0, 100)
+                    quantity = randint(6, 28) * items
+
+                    actual_cost = randint(100, 200) * quantity * 0.01
+
+                    # We don't care about net_cost.
+                    net_cost = actual_cost
+
+                    row = [
+                        'sha',  #  This value doesn't matter.
+                        practice.ccg.regional_team_id,
+                        practice.ccg.stp_id,
+                        practice.ccg_id,
+                        practice.code,
+                        bnf_code,
+                        bnf_name,
+                        items,
+                        net_cost,
+                        actual_cost,
+                        quantity,
+                        timestamp,
+                    ]
+
+                    prescribing_rows.append(row)
+
+        # In production, normalised_prescribing_standard is actually a view,
+        # but for the tests it's much easier to set it up as a normal table.
+        table = Client('hscic').get_or_create_table(
+            'normalised_prescribing_standard', schemas.PRESCRIBING_SCHEMA
+        )
+
+        # Upload prescribing_rows to normalised_prescribing_standard, and
```

Comment:
This is copy-and-paste from earlier, could be a function?

Reply:
Good idea.